fah_alchemy/cli.py

```diff
@@ -364,30 +372,115 @@ def reset(url, user, password, dbname):
+    if identity_type == "user":
+        identity_type_cls = CredentialedUserIdentity
+    elif identity_type == "compute":
+        identity_type_cls = CredentialedComputeIdentity
+    else:
+        raise RuntimeError(f"Unknown identity type {identity_type}")
+
+    return identity_type_cls
+
+
+def identity_type(func):
+    identity_type = click.option(
+        "--identity-type",
+        '-t',
+        default="user",
+        help="User type",
+        type=click.Choice(["user", "compute"], case_sensitive=False),
+    )
+    return identity_type(func)
+
+
+def identity_params(func):
+    identity = click.option("--identifier", '-i', help="identifier", required=True, type=str)
+    key = click.option("--key", '-k', help="key", required=True, type=str)
+    return identity(key((func)))
+
+
 @cli.group()
-def user():
+def identity():
     ...
 
 
-@user.command()
-def add():
-    """Add a user to the database."""
-    ...
+@identity.command()
+@db_params
+@identity_type
+@identity_params
+def add(url, user, password, dbname, identity_type, identifier, key):
+    """Add a credentialed identity to the database.
+
+    """
+    from .storage.statestore import get_n4js
+    from .settings import Neo4jStoreSettings
+
+    cli_values = url | user | password | dbname
+
+    settings = get_settings_from_options(cli_values, Neo4jStoreSettings)
+    n4js = get_n4js(settings)
+
+    identity_type_cls = _identity_type_string_to_cls(identity_type)
+    identity_model = identity_type_cls(hashed_key=hash_key(key), identifier=identifier)
+    n4js.create_credentialed_entity(identity_model)
+
+
+@identity.command()
+@db_params
+@identity_type
+def list(url, user, password, dbname, identity_type):
+    """List all credentialed entities of the given type.
+
+    """
+    from .storage.statestore import get_n4js
+    from .settings import Neo4jStoreSettings
+
+    cli_values = url | user | password | dbname
+
+    settings = get_settings_from_options(cli_values, Neo4jStoreSettings)
+    n4js = get_n4js(settings)
+
+    identity_type_cls = _identity_type_string_to_cls(identity_type)
+    click.echo(n4js.list_credentialed_entities(identity_type_cls))
+
+
+@identity.command()
+@db_params
+@identity_type
+@identity_params
+def remove(url, user, password, dbname, identity_type, identifier, key):
+    """Remove a credentialed identity from the database.
+
+    """
+    from .storage.statestore import get_n4js
+    from .settings import Neo4jStoreSettings
+
+    cli_values = url | user | password | dbname
+
+    settings = get_settings_from_options(cli_values, Neo4jStoreSettings)
+    n4js = get_n4js(settings)
+
+    identity_type_cls = _identity_type_string_to_cls(identity_type)
+    if authenticate(n4js, identity_type_cls, identifier, key):
+        n4js.remove_credentialed_identity(identifier, identity_type_cls)
+    else:
+        raise AuthenticationError("Authentication with StateStore failed.")
```

Comment:
An admin that needs to remove an identity from a deployed `fah-alchemy` instance generally won't have the credential for that identity, e.g. a user's key. We should remove `key` as an argument from this CLI command.

Reply:
Roger that, this was due to my misconception as to who would be using the CLI and where. 👍 Makes perfect sense.

Reply:
No problem!